docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,596 @@
+            else:
+                continue  # skip this item if it has no name
+
+            item_quantity_str: str | None = redis_client.hget(
+                f"item_id:{item_id}", "quantity"
+            )
+            if item_quantity_str is not None:
+                item_quantity: int = int(item_quantity_str)
+            else:
+                item_quantity = 0
+
+            items.append(
+                ItemPayload(item_id=item_id, item_name=item_name, quantity=item_quantity)
+            )
+
+        return {"items": items}
+
+
+    # Route to delete a specific item by id but using Redis
+    @app.delete("/items/{item_id}")
+    def delete_item(item_id: int) -> dict[str, str]:
+        if not redis_client.hexists(f"item_id:{item_id}", "item_id"):
+            raise HTTPException(status_code=404, detail="Item not found.")
+        else:
+            item_name: str | None = redis_client.hget(f"item_id:{item_id}", "item_name")
+            redis_client.hdel("item_name_to_id", f"{item_name}")
+            redis_client.hdel(f"item_id:{item_id}", "item_id")
+            return {"result": "Item deleted."}
+
+
+    # Route to remove some quantity of a specific item by id but using Redis
+    @app.delete("/items/{item_id}/{quantity}")
+    def remove_quantity(item_id: int, quantity: int):
+        if not redis_client.hexists(f"item_id:{item_id}", "item_id"):
+            raise HTTPException(status_code=404, detail="Item not found.")
+
+        item_quantity: str | None = redis_client.hget(f"item_id:{item_id}", "quantity")
+
+        # if quantity to be removed is higher or equal to item's quantity, delete the item
+        if item_quantity is None:
+            existing_quantity: int = 0
+        else:
+            existing_quantity: int = int(item_quantity)
+        if existing_quantity <= quantity:
+            item_name: str | None = redis_client.hget(f"item_id:{item_id}", "item_name")
+            redis_client.hdel("item_name_to_id", f"{item_name}")
+            redis_client.hdel(f"item_id:{item_id}", "item_id")
+            return {"result": "Item deleted."}
+        else:
+            redis_client.hincrby(f"item_id:{item_id}", "quantity", -quantity)
+            return {"result": f"{quantity} items removed."}
+    ```
+
+12.	Re-run the debugger to test this application by interacting with the `/docs` route.
+
+## Optional: Set up database deletion
+Because our data will be now be persisted by redis, you may want to create a script that will allow you to erase all testing data. For that you can create a new file called `flushdb.py` with the following content:
+```python
+import redis
+
+redis_client = redis.StrictRedis(host='0.0.0.0', port=6379, db=0, decode_responses=True)
+redis_client.flushdb()
+
+```
+Then when you want to reset your database, you can open the flushdb.py file in VS Code and click on the run button on the top-right corner of the editor or run the **Python: Run Python File in Terminal** command.
```

Comment:
I got the following error when I tried to run flushdb.py. Maybe because the venv isn't selected, but when I try to select the venv, it isn't in the list of available interpreters.

```text
vscode ➜ /workspaces/groceries-plugin $ /usr/local/bin/python /workspaces/groceries-plugin/flushdb.py
Traceback (most recent call last):
  File "/workspaces/groceries-plugin/flushdb.py", line 1, in <module>
    import redis
ModuleNotFoundError: No module named 'redis'
```

![image](https://github.com/luabud/vscode-docs/assets/12820898/79e7dbc1-49d3-48f8-9e20-f2b8a3313c30)

![image](https://github.com/luabud/vscode-docs/assets/12820898/f48a2b1a-1a36-46a6-89c3-b7cf017d5033)